InnerEye/ML/deep_learning_config.py

```diff
@@ -183,39 +183,184 @@ def add_subfolder(self, subfolder: str) -> DeepLearningFileSystemConfig:
-                                                  "cost, hence good for use in unit testing. However, it "
-                                                  "does not give the same result as running with 1 worker process)")
-    shuffle: bool = param.Boolean(True, doc="If true, the dataset will be shuffled randomly during training.")
-    num_epochs: int = param.Integer(100, bounds=(1, None), doc="Number of epochs to train.")
-    start_epoch: int = param.Integer(0, bounds=(0, None), doc="The first epoch to train. Set to 0 to start a new "
-                                                              "training. Set to a value larger than zero for starting"
-                                                              " from a checkpoint.")
+    number_of_cross_validation_splits: int = param.Integer(0, bounds=(0, None),
+                                                           doc="Number of cross validation splits for k-fold cross "
+                                                               "validation")
+    cross_validation_split_index: int = param.Integer(DEFAULT_CROSS_VALIDATION_SPLIT_INDEX, bounds=(-1, None),
+                                                      doc="The index of the cross validation fold this model is "
+                                                          "associated with when performing k-fold cross validation")
+    perform_training_set_inference: bool = \
+        param.Boolean(False,
+                      doc="If False (default), run full image inference on validation and test set after training. If "
+                          "True, also run full image inference on the training set")
+    perform_validation_and_test_set_inference: bool = \
+        param.Boolean(True,
+                      doc="If True (default), run full image inference on validation and test set after training.")
+    weights_url: str = param.String(doc="If provided, a url from which weights will be downloaded and used for model "
+                                        "initialization.")
+    local_weights_path: Optional[Path] = param.ClassSelector(class_=Path,
+                                                             default=None,
+                                                             allow_None=True,
+                                                             doc="The path to the weights to use for model "
+                                                                 "initialization, when training outside AzureML.")
+    generate_report: bool = param.Boolean(default=True,
+                                          doc="If True (default), write a modelling report in HTML format. If False,"
+                                              "do not write that report.")
+    # The default multiprocessing start_method in both PyTorch and the Python standard library is "fork" for Linux and
+    # "spawn" (the only available method) for Windows. There is some evidence that using "forkserver" on Linux
```

Comment:
If forkserver reduces the chances of stuck jobs why is not the default?

Reply:
Honestly I don't know. This is an ages old comment from David who was battling with stuck jobs, the comment was copied verbatim (these params were only moved out of DeepLearningConfig, but are otherwise unchanged)